src/components/rating/rating.tsx

```diff
@@ -222,11 +221,48 @@ export class Rating implements LabelableComponent, FormComponent, InteractiveCom
     this.setFocus();
   }
 
+  /**
+   * update `value` property of the rating component
+   *
+   * @param value value used to update `value` property
+   * @internal
+   */
   private updateValue(value: number) {
     this.value = value;
     this.calciteRatingChange.emit({ value });
   }
 
+  /**
+   * register keyboard input and reset `value` property to `0` if user enters 'Space' or 'Enter'
+   *
+   * @param event keyboard event
+   * @internal
+   */
+  private onKeyboardPressed(event: KeyboardEvent): void {
+    // click is fired from the the component's input, so we treat this as an internal event
+    event.stopPropagation();
+    event.preventDefault();
+    if (event.key === "Enter" || event.code === "Space") {
```

Comment:
Can you use `event.key === " "` instead of `event.code === "Space"`?

Reply:
@driskull  I have implemented your suggestions in new updates